src/Language/DifferentialDatalog/Pos.hs

```diff
@@ -50,7 +52,24 @@ spos x = let (s,e) = pos x
          in sourceName s ++ ":" ++ (show $ sourceLine s) ++ ":" ++ (show $ sourceColumn s) ++ "-"
                                 ++ (show $ sourceLine e) ++ ":" ++ (show $ sourceColumn e)
 
-nopos::Pos 
+sposFragment :: (WithPos a) => a -> M.Map String String -> String -> String
```

Comment:
A comment here would be good, perhaps even including an example output produced by this function.